chapters/annotations.tex

```diff
@@ -68,27 +73,190 @@ \section{Annotations for Documentation}\doublelabel{annotations-for-documentatio
+mandatory, as every figure needs a title for presentation in contexts such as a
+list of figures, and as it is generally hard for tools to automatically generate
+a meaningful title.  On the other hand, when the \lstinline!title! of a
+\lstinline!Plot! isn't provided, the tool produces a default, but the default is
+allowed to be empty.  Providing the empty string as \lstinline!title! of a
+\lstinline!Plot! means that no title should be shown.  The plot title is not to
+be confused with the plot \emph{label} which is never empty, see below. Variable
+replacements, as described in \autoref{variable-replacements}, can be used in the
+\lstinline!title! of \lstinline!Figure! and \lstinline!Plot!.
+
+The \lstinline!identifier! in \lstinline!Figure! and \lstinline!Plot! is an
+optional \lstinline!String! identifier, and is intended to identify the
+\lstinline!Figure! and \lstinline!Plot! for programmatic access.  A
+\lstinline!Figure! must be uniquely identified by its \lstinline!identifier! and
+containing class.  This means that a \lstinline!Figure! \lstinline!identifier!
+must be unique among all \lstinline!Figure! annotations within the same
+\lstinline!figures! annotation as well as among all \lstinline!figures!
+annotations from inherited classes.  A \lstinline!Plot! \lstinline!identifier!
+on the otherhand is only required to be unique among the \lstinline!plots! in
+the the same \lstinline!Figure! annotation.
+
+\begin{nonnormative}
+For \lstinline!Figure!, this makes it possible to reference the plot from a
+tool-specific scripting environment. For \lstinline!Plot!, this makes it
+possible to reference the plot in the figure caption, which becomes useful when
+the \lstinline!Figure! contains more than one \lstinline!Plot!.
+\end{nonnormative}
+
+Every \lstinline!Plot! has an automatically generated \emph{label} which is
+required to be shown as soon as at least one \lstinline!Plot! in the
+\lstinline!Figure! has an \lstinline!identifier!.  A tool is free to choose both
+labeling scheme (such as \emph{a}, \emph{b}, \dots, or \emph{i}, \emph{ii}, \dots) placement in the plot,
+and styling in the plot itself as well as in other contexts.
+
+When a \lstinline!Figure! defines a non-empty \lstinline!group!, it is used to
+organize figures similar to how \lstinline!group! is used in the
+\lstinline!Dialog! annotation (see \autoref{annotations-for-the-graphical-user-interface}).  However, leaving \lstinline!group! at
+the default of an empty string does not mean that a group will be created
+automatically, but that the figure resides outside of any group. The
+\lstinline!group! is both the key used for grouping, and the name of the group
+for display purposes.
+
+If the \lstinline!preferred! attribute of \lstinline!Figure! is \lstinline!true!
+tools should should show the figure automatically when the class is simulated. A
+class may define multiple \lstinline!preferred! figures.
```

Comment:
I think we should start by writing the normative part so that it doesn't require things that don't always make sense, and hopefully this removes the need for a non-normative elaboration.  How about this?
> The `preferred` attribute of `Figure` indicates whether the figure should be given preference when automatically determining which figures to show.  For example, a tool might choose to automatically show all preferred figures when the class is simulated.

Reply:
Sounds great to me!

Reply:
Fixed.  Resolving.